pkg/tagger/utils/expiry.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package utils
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+// Expire regularly pools the source keeping the elements reporting.
+// Currently only used for ECSCollector and active containers.
+// It keeps an internal state to only send the updated elements (only used for containers to start with).
+type Expire struct {
+	sync.Mutex
+	expiryDuration time.Duration
+	lastSeen       map[string]time.Time
+}
+
+// NewExpire creates a new Expire object. Called when a Collector is started.
+// Only used for the ECS collector to start with.
+func NewExpire(expiryDuration time.Duration) (*Expire, error) {
+	if expiryDuration <= 0 {
+		return nil, errors.New("expiryDuration must be above 0")
+	}
+	return &Expire{
+		expiryDuration: expiryDuration,
+		lastSeen:       make(map[string]time.Time),
+	}, nil
+}
+
+// Update will update the map of the Expire obect with the elements and the time they reported.
+// Will return True if the element passed (container) is not found in the current lastseen map.
```

Comment:
no need to expose class internal mechanism, just `if the entity name passed as not been passed before`

Reply:
I removed the details on the mechanism - Though, Update updates the timestamp if the entity has been seen before on top of adding it if it has not been seen before.